src/vm/methodtablebuilder.cpp

```diff
@@ -1489,19 +1489,19 @@ MethodTableBuilder::BuildMethodTableThrowing(
         }
     }
 
-#if defined(_TARGET_X86_) || defined(_TARGET_AMD64_) || defined(_TARGET_ARM64_)
+#if defined(FEATURE_HW_INTRINSICS) && (defined(_TARGET_XARCH_) || defined(_TARGET_ARM64_))
     if (GetModule()->IsSystem() && !bmtGenerics->HasInstantiation())
     {
         LPCUTF8 className;
         LPCUTF8 nameSpace;
         HRESULT hr = GetMDImport()->GetNameOfTypeDef(bmtInternal->pType->GetTypeDefToken(), &className, &nameSpace);
 
 #if defined(_TARGET_ARM64_)
-        // All the funtions in System.Runtime.Intrinsics.Arm.Arm64 are hardware intrinsics.
+        // All the funtions in System.Runtime.Intrinsics.Arm.Arm64 are hardware intrinsics.	        if ((hr == S_OK) && (strncmp(nameSpace, "System.Runtime.Intrinsics", 25) == 0))
```

Comment:
Need to delete the added code at the end of this comment

Reply:
Fixed.